src/App.tsx

```diff
@@ -1,24 +1,291 @@
+  removeTodo,
+  addTodoToServer,
+  changeTodo,
+} from './api/todos';
+import { TodosList } from './components/TodosList/TodosList';
+import { Todo } from './types/Todo';
+import { ErrorInfo } from './components/ErrorInfo/ErrorInfo';
+import {
+  preparedTodos,
+  getcompletedTodosIds,
+  filterTodosByCompletion,
+} from './utils/todoUtils';
+import { StatusValue } from './types/StatusValue';
 
-const USER_ID = 0;
+const USER_ID = 10725;
+
+export const App: FC = () => {
+  const formRef = useRef<HTMLInputElement | null>(null);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [queryTodo, setQueryTodo] = useState('');
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [statusTodo, setstatusTodo] = useState<StatusValue>(StatusValue.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [visibleError, setVisibleError] = useState('');
+
+  const fetchTodosFromServer = async () => {
+    try {
+      const fetchedTodos = await getTodos(USER_ID);
+
+      setTodos(fetchedTodos as Todo[]);
+    } catch (error) {
+      setVisibleError('Unable to load a todos');
+    }
+  };
+
+  useEffect(() => {
+    if (formRef.current) {
+      formRef.current.focus();
+    }
+
+    fetchTodosFromServer();
+  }, [tempTodo, loadingTodoIds, visibleError]);
+
+  const completedTodosIds = useMemo(() => (
+    getcompletedTodosIds(todos)
+  ), [todos]);
+
+  const visibleTodos = useMemo(() => (
+    preparedTodos(todos, statusTodo)
+  ), [todos, statusTodo]);
+
+  const removeTodos = useCallback(async (todoIds: number[]) => {
+    try {
+      setLoadingTodoIds(prevIds => [...prevIds, ...todoIds]);
+
+      await Promise.all(
+        todoIds.map(async id => {
+          await removeTodo(id);
+        }),
+      );
+
+      const updatedTodos = todos.filter(todo => !todoIds.includes(todo.id));
+
+      setTodos(updatedTodos);
+    } catch (error) {
+      setVisibleError('Unable to delete a todo');
+    } finally {
+      setLoadingTodoIds([]);
+    }
+  }, [todos]);
+
+  const handleToggleTodoStatus = useCallback(async (
+    todoIds: number[],
+  ) => {
+    try {
+      setLoadingTodoIds((prevIds) => [...prevIds, ...todoIds]);
+
+      await Promise.all(
+        todoIds.map(async id => {
+          const todoToUpdate = todos.find(todo => todo.id === id);
+
+          if (todoToUpdate) {
+            const updatedTodo = await changeTodo(id, {
+              ...todoToUpdate,
+              completed: !todoToUpdate.completed,
+            });
+
+            setTodos(prevTodos => prevTodos.map(todo => (
+              todo.id === updatedTodo.id
+                ? updatedTodo
+                : todo
+            )));
+          }
+        }),
+      );
+    } catch (error) {
+      setVisibleError('Unable to update a todo');
+    } finally {
+      setLoadingTodoIds([]);
+    }
+  }, [todos]);
+
+  const changeTitle = useCallback(async (
```

Comment:
That's the same logic. Can't we handle both cases in a single function like `handleUpdate`?